Plugins/SwiftGenPlugin/Plugin.swift

```diff
@@ -31,6 +31,33 @@ struct SwiftGenPlugin: BuildToolPlugin {
   }
 }
 
+#if canImport(XcodeProjectPlugin)
+import XcodeProjectPlugin
+
+extension SwiftGenPlugin: XcodeBuildToolPlugin {
+  func createBuildCommands(context: XcodePluginContext, target: XcodeTarget) throws -> [Command] {
+    let fileManager = FileManager.default
+
+    // Possible paths where there may be a config file (root of package, target dir.)
+    let configurations: [Path] = [context.xcodeProject.directory]
+      .map { $0.appending("swiftgen.yml") }
+      .filter { fileManager.fileExists(atPath: $0.string) }
```

Comment:
In the current implementation, it cares only about a top-level configuration.
We have to care about a target-specified configuration.

Unfortunately, we can't fetch the top-level directory of each target from `XcodeBuildToolPlugin`, because a top-level directory of targets doesn't always exist in xcodeproj.

But most likely it will be `$SRCROOT/TARGET_NAME`.

So it's better to add `target.displayName` as a search path.

```suggestion
    let configurations: [Path] = [
        context.xcodeProject.directory,
        context.xcodeProject.directory.appending(subpath: target.displayName),
    ]
      .map { $0.appending("swiftgen.yml") }
      .filter { fileManager.fileExists(atPath: $0.string) }
```

Reply:
This works well for my multi-target situation, thanks!